packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -356,13 +346,57 @@ export class FieldSchema<
 		 */
 		public readonly props?: FieldProps<TCustomMetadata>,
 	) {
+		if (!(this instanceof FieldSchemaAlpha)) {
+			throw new UsageError("FieldSchema is @sealed: sub-classing is not allowed.");
```

Comment:
Nit: Probably worth tacking a `@privateRemarks` comment on the constructor noting that it only allows being extended by `FieldSchemaAlpha`, and any new internal implementations will need to update the check here.

Reply:
new internal implementations would extend FieldSchemaAlpha, and not update this. Updating this instead would make down casts to FieldSchemaAlpha invalid.

I have added some (public facing) docs to better clear up things. Hopefully the requirements for internal use will be implied by those.